modules/planning/tasks/deciders/path_assessment_decider.cc

```diff
@@ -0,0 +1,69 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/planning/tasks/deciders/path_assessment_decider.h"
+
+namespace apollo {
+namespace planning {
+
+using apollo::common::ErrorCode;
+using apollo::common::Status;
+
+PathAssessmentDecider::PathAssessmentDecider(const TaskConfig& config)
+    : Decider(config) {}
+
+Status PathAssessmentDecider::Process(
+    Frame* const frame, ReferenceLineInfo* const reference_line_info) {
+  // Sanity checks.
+  CHECK_NOTNULL(frame);
+  CHECK_NOTNULL(reference_line_info);
+
+  // Check the validity of paths (the optimization output).
+  // 1. First check the regular path's validity.
+  const PathData* regular_path_data = reference_line_info->mutable_path_data();
```

Comment:
mutable_xxx is a pointer and potentially can be updated. 
here you can just use the reference_line_info()->path_data(). 
the same applies to fallback_path_data.

Reply:
This is because in the second stage(to be implemented later), path_data will be changed. Some in-lane/out-lane info will be updated to path_data for speed_decider to use.

Reply:
The practice is:
1. Use const reference if you don't need to change an object.
2. Use pointer if you need to change it.

So here should be `const PathData& regular_path_data = reference_line_info->path_data();`
The `mutable_xx()` interface is used when you want to change it, so it never shows along with a `const`.

Reply:
Sorry I put the "const" by mistake. I deleted that.

Reply:
both regular_path_data and fallback_path_data might be modified later in the same function. I plan to implement those at after line 54.